### PR TITLE
feat(1.6): add License.Acknowledgement

### LINKED ISSUE #166: Add `License.Acknowledgement`
This is to track the addition of the `"acknowledgement"` field to type `License` (spec 1.6).

See:
* https://cyclonedx.org/docs/1.6/json/#components_items_licenses_oneOf_i0_items_license_acknowledgement
* https://github.com/CycloneDX/specification/pull/408

### PR DESCRIPTION
**Pull request by mcombuechen**
This adds `License.Acknowledgement` for spec 1.6 support.

Closes #166.

### CODE REVIEW

File: cyclonedx.go [hunk 1]
```diff
@@ -91,7 +91,7 @@ type BOM struct {
 	Vulnerabilities    *[]Vulnerability     `json:"vulnerabilities,omitempty" xml:"vulnerabilities>vulnerability,omitempty"`
 	Annotations        *[]Annotation        `json:"annotations,omitempty" xml:"annotations>annotation,omitempty"`
 	Formulation        *[]Formula           `json:"formulation,omitempty" xml:"formulation>formula,omitempty"`
-	Definitions        *Definitions         `json:"definitions" xml:"definitions,omitempty"`
+	Definitions        *Definitions         `json:"definitions,omitempty" xml:"definitions,omitempty"`
```
Comment (mcombuechen) [inline cyclonedx.go:94]: Unrelated fix for something that I missed in #158.

**Review by nscuro — APPROVED** [review]
